src/solvers/dgmulti/dg.jl

```diff
@@ -485,6 +489,55 @@ function calc_single_boundary_flux!(cache, t, boundary_condition, boundary_key,
   # However, we don't have to re-reshape, since cache.flux_face_values still retains its original shape.
 end
 
+function calc_single_boundary_flux!(cache, t, boundary_condition, boundary_key,
+                                    have_nonconservative_terms::True,
+                                    mesh, equations, dg::DGMulti{NDIMS}) where {NDIMS}
+
+  rd = dg.basis
+  md = mesh.md
+  (; u_face_values, flux_face_values) = cache
+  (; xyzf, nxyzJ, Jf) = md
+  surface_flux, nonconservative_flux = dg.surface_integral.surface_flux
+
+  # reshape face/normal arrays to have size = (num_points_on_face, num_faces_total).
+  # mesh.boundary_faces indexes into the columns of these face-reshaped arrays.
+  num_pts_per_face = rd.Nfq ÷ rd.Nfaces
+  num_faces_total = rd.Nfaces * md.num_elements
```

Comment:
```suggestion
  num_pts_per_face = rd.Nfq ÷ StartUpDG.num_faces(rd.element_type)
  num_faces_total = StartUpDG.num_faces(rd.element_type) * md.num_elements
```
For some reason, these are not type stable otherwise.

Reply:
I must have missed these - I just fixed a few similar ones in https://github.com/trixi-framework/Trixi.jl/pull/1419 recently. 

This seems to be due to StartUpDG.jl overloading `getproperty` for `rd.Nfaces`, which isn't type stable. I think I've removed all instances of this in Trixi.jl now.